antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -201,6 +224,13 @@ def append_commands(
         command_objs = self._check_commands_validity(study_id, commands)
         validated_commands = transform_command_to_dto(command_objs, commands)
         first_index = len(study.commands)
+
+        # type check in case user is None
+        if not params.user or not params.user.id:
```

Comment:
I belive you can't reach this code as you check above that the user has the right to edit the study and if the user is None we crash here. I'll let you check but I think you can remove this if